pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -268,24 +268,24 @@ func (pl *dynamicResources) EventsToRegister() []framework.ClusterEventWithHint
 func (pl *dynamicResources) podResourceClaims(pod *v1.Pod) ([]*resourcev1alpha2.ResourceClaim, error) {
 	claims := make([]*resourcev1alpha2.ResourceClaim, 0, len(pod.Spec.ResourceClaims))
 	for _, resource := range pod.Spec.ResourceClaims {
-		claimName := resourceclaim.Name(pod, &resource)
-		isEphemeral := resource.Source.ResourceClaimTemplateName != nil
-		claim, err := pl.claimLister.ResourceClaims(pod.Namespace).Get(claimName)
+		claimName, mustCheckOwner, err := resourceclaim.Name(pod, &resource)
+		if err != nil {
+			return nil, err
+		}
+		if claimName == nil {
+			// Not needed, continue without it.
+			continue
+		}
+		claim, err := pl.claimLister.ResourceClaims(pod.Namespace).Get(*claimName)
 		if err != nil {
-			// The error usually has already enough context ("resourcevolumeclaim "myclaim" not found"),
-			// but we can do better for generic ephemeral inline volumes where that situation
-			// is normal directly after creating a pod.
-			if isEphemeral && apierrors.IsNotFound(err) {
-				err = fmt.Errorf("waiting for dynamic resource controller to create the resourceclaim %q", claimName)
-			}
```

Comment:
I don't have a whole lot of context on why this was needed to begin with, but why is this not needed anymore? I guess it's all captured within the `resourceclaim.Name()` call now?

Reply:
Correct. It'll check for this based on the pod status field.

Reply:
Ack